python/tests/integration/arcticdb/test_arctic_batch.py

```diff
@@ -1299,3 +1302,51 @@ def test_read_description_batch_empty_nat(arctic_library):
     for sym in range(num_symbols):
         assert np.isnat(results_list[sym].date_range[0]) == True
         assert np.isnat(results_list[sym].date_range[1]) == True
+
+
+def test_mutlithread_read(library_factory):
+    set_config_int("VersionStore.NumCPUThreads", 2)
+    set_config_int("VersionStore.NumIOThreads", 2)
+    num_days = 2
+    num_symbols = 10
+    dt = datetime(2019, 4, 8, 0, 0, 0)
+    num_columns = 10
+    num_rows_per_day = 5
+    write_requests = []
+    read_requests = []
+    list_dataframes = {}
+    columns = random_strings_of_length(num_columns, num_columns, True)
+    df = generate_dataframe(random.sample(columns, num_columns), dt, num_days, num_rows_per_day)
+    for sym in range(num_symbols):
+        write_requests.append(WritePayload("sym_" + str(sym), df, metadata="great_metadata" + str(sym)))
+        read_requests.append("sym_" + str(sym))
+        list_dataframes[sym] = df
+    lib = library_factory(LibraryOptions(rows_per_segment=10, dedup=True))
+    lib.write_batch(write_requests)
+    q = queue.Queue()
+
+    def read_batch():
+        try:
+            read_batch_result = lib.read_batch(read_requests)
+            for sym in range(num_symbols):
+                assert read_batch_result[sym].metadata == "great_metadata" + str(sym)
+                assert_frame_equal(read_batch_result[sym].data, list_dataframes[sym])
+        except Exception as e:
+            q.put(e)
+
+    def read():
+        try:
+            for sym in range(num_symbols):
+                assert_frame_equal(lib.read("sym_" + str(sym)).data, list_dataframes[sym])
+        except Exception as e:
+            q.put(e)
+
+    read_th_count = 1
+    batch_read_th_count = 1
```

Comment:
Increase these thread counts?

What is the point of the queue (which isn't synchronised correctly), why not just let the exception bubble up and fail the test?

Reply:
I can increase the thread count. Yet according to my test, current setting is sufficient to repeating the issue. I would like to keep the additional time needed for running the test as minimum as possible.

Reply:
Sounds good, the minimal amount to repro the issue is fine.